dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,293 @@
+                The number of features to select.
+            strict:
+                If ``False`` the constraint on the number of selected features
+                is ``<=`` rather than ``==``.
+
+        Returns:
+            A constrained quadratic model.
+
+        .. [Milne et al.] Milne, Andrew, Maxwell Rounds, and Phil Goddard. 2017. "Optimal Feature
+            Selection in Credit Scoring and Classification Using a Quantum Annealer."
+            1QBit; White Paper.
+            https://1qbit.com/whitepaper/optimal-feature-selection-in-credit-scoring-classification-using-quantum-annealer
+        """
+
+        X = np.atleast_2d(np.asarray(X))
+        y = np.asarray(y)
+
+        if X.ndim != 2:
+            raise ValueError("X must be a 2-dimensional array-like")
+
+        if y.ndim != 1:
+            raise ValueError("y must be a 1-dimensional array-like")
+
+        if not 0 <= alpha <= 1:
+            raise ValueError(f"alpha must be between 0 and 1, given {alpha}")
+
+        if num_features <= 0:
+            raise ValueError(f"num_features must be a positive integer, given {num_features}")
+
+        cqm = dimod.ConstrainedQuadraticModel()
+        cqm.add_variables(dimod.BINARY, X.shape[1])
+
+        # add the k-hot constraint
+        cqm.add_constraint(((v, 1) for v in cqm.variables), '==' if strict else '<=', num_features)
+
+        with tempfile.TemporaryFile() as fX, tempfile.TemporaryFile() as fout:
+            # we make a copy of X because we'll be modifying it in-place within
+            # some of the functions
+            X_copy = np.memmap(fX, X.dtype, mode="w+", shape=(X.shape[0], X.shape[1] + 1))
+            X_copy[:, :-1] = X
+            X_copy[:, -1] = y
+
+            # make the matrix that will hold the correlations
+            correlations = np.memmap(
+                fout,
+                dtype=np.result_type(X, y),
+                mode="w+",
+                shape=(X_copy.shape[1], X_copy.shape[1]),
+                )
+
+            # main calculation. It modifies X in-place
+            corrcoef(X_copy, out=correlations, rowvar=False, copy=False)
+
+            # we don't care about the direction of correlation in terms of
+            # the penalty/quality
+            np.absolute(correlations, out=correlations)
+
+            # our objective
+            np.fill_diagonal(correlations, -correlations[:, -1] * alpha)
+
+            # Note: the full symmetric matrix (with both upper- and lower-diagonal
+            # entries for each correlation coefficient) is retained for consistency with
+            # the original formulation from Milne et al.
+            it = np.nditer(correlations[:-1, :-1], flags=['multi_index'], op_flags=[['readonly']])
+            cqm.set_objective((*it.multi_index, x) for x in it)
+
+        return cqm
+
+    def fit(
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: typing.Optional[float] = None,
```

Comment:
Because this is a hyper-parameter I'm not sure it should be passable in the fit function, in general hyperparameters in scikit are passed at instantiation. Not universal, but a pattern.

Reply:
Here we're allowing the user to override the value specified in instantiation. Below we do
```python
if alpha is None:
    alpha = self._alpha
```
So we get both patterns. It felt nice to be able to override it locally.